### PR TITLE
(PA-2426) Leatherman on OSX

### PR DESCRIPTION
**Pull request by denis-susan**
- Removed pin for boost on all osx platforms
- Added compiler flags for building rapidjson on osx 10.14 (apple clang 10 complains about expansion-to-defined and delete-non-virtual-destructor)
- changed osx-10.13 platform to build on osx-10.13 vm
- changed osx-10.14 platform to build on osx-10.14 vm

### CODE REVIEW

**Comment by gimmyxd** [comment]
jenkins please test this on osx1012-64a osx1013-64a osx1014-64a

**Comment by gimmyxd** [comment]
jenkins please test this on osx1012-64a, osx1013-64a, osx1014-64a

**Comment by gimmyxd** [comment]
jenkins please test this on osx1012-64a,osx1013-64a,osx1014-64a

**Comment by denis-susan** [comment]
jenkins please test this on osx1012-64a,osx1013-64a,osx1014-64a

File: configs/components/cpp-hocon.rb [hunk 1]
```diff
@@ -11,7 +11,12 @@
   if platform.is_macos?
     toolchain = ""
     cmake = "/usr/local/bin/cmake"
-    special_flags = "-DCMAKE_CXX_FLAGS='#{settings[:cflags]}'"
+    boost_static_flag = "-DBOOST_STATIC=OFF"
```
Comment (branan) [inline configs/components/cpp-hocon.rb]: If we're turning off static boost, shouldn't we have to copy the `.dylib` files into our lib directory to ensure they'll be available on end-user systems?

Comment (denis-susan) [reply configs/components/cpp-hocon.rb]: Added boost in puppet-runtime: [142](https://github.com/puppetlabs/puppet-runtime)

**Comment by denis-susan** [comment]
Depends on: [vanagon/598](https://github.com/puppetlabs/vanagon/pull/598)

**Comment by mwaggett** [comment]
vanagon 0.15.21 should fix the signing issue on osx-10.14

**Comment by denis-susan** [comment]
jenkins please test this on osx1012-64a,osx1013-64a,osx1014-64a

**Comment by denis-susan** [comment]
@mwaggett same issue as before, can you check it out?

**Comment by mwaggett** [comment]
jenkins please test this on osx1012-64a,osx1013-64a,osx1014-64a

**Comment by mwaggett** [comment]
oh, it only reruns the failed bits

**Comment by mwaggett** [comment]
jenkins please test this on osx1012-64a,osx1013-64a,osx1014-64a

**Comment by mwaggett** [comment]
welp @denis-susan good news is the signing issue is resolved (it was using an old version of vanagon), bad news is I've got new failures for you

**Comment by denis-susan** [comment]
jenkins please test this on osx1012-64a,osx1013-64a,osx1014-64a

**Comment by denis-susan** [comment]
jenkins please test this on osx1012-64a,osx1013-64a,osx1014-64a

**Comment by denis-susan** [comment]
successful test run: https://jenkins-platform.delivery.puppetlabs.net/job/platform_puppet-agent-extra_puppet-agent-integration-suite_pr/908/

**Pull request merged**